code/game/machinery/ai_relay.dm

```diff
@@ -0,0 +1,176 @@
+GLOBAL_LIST_EMPTY(ai_relays)
+
+#define AI_RELAY_HACK_TIME 600
+#define FLIP_DELAY 150
+
+/obj/machinery/ai_relay
+	name = "processing relay"
+	desc = "A mighty piece of hardware used to process massive amounts of data. It allows the AI to interface with various machines on the station."
+
+	icon = 'icons/obj/machines/telecomms.dmi'
+	icon_state = "hub"
+
+	critical_machine = TRUE
+	density = TRUE
+
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 500
+	active_power_usage = 5000 //10x
+
+	var/on = FALSE
+
+	var/list/linked_ais = list()
```

Comment:
You could potentially optimise this to be a 2-d dictionary like:

var/list/ais = list()

ais[hacked] += a_hacked_ai_ref
ais[vanilla] += an_ai_ref

Reply:
Good idea

Reply:
I'm ditching this idea, it looked good on the surface, but it's not very practical